wordpress-mobile/WordPress-iOS
# Show a loading indicator when loading images/gifs

Fixes #9303 

This PR adds a simple activity indicator to `CachedAnimatedImageView`, and it's controlled by the `ImageLoader` class.

In this way, we will get a loading indicator everywhere the `ImageLoader` class is used for free.

This implementation also supports the use of a custom view as a loading indicator with style `centered` and `full view`.

![images_loading_indicator](https://user-images.githubusercontent.com/9772967/39786379-bb1ba7f8-52f6-11e8-9249-8e81fe08c80f.png)

To test:

- Visit the Reader list and check if the default loading indicator shows properly.
- Visit the Post list view and check if the default loading indicator shows properly.

With a small code addition, you can test a custom loading indicator (as shown in the images).

In the `ImageLoader.swift` file, at the top of the file, add:
```swift
extension MediaCellProgressView: ActivityIndicatorType {
    func startAnimating() {
        isHidden = false
        state = .indeterminate
    }

    func stopAnimating() {
        isHidden = true
        state = .stopped
    }
}
```

Then, in the same file, after the line 29 (at the end of the `init()` method) add this for a full view example:
```swift
let loadingView = MediaCellProgressView(frame: .zero)
imageView.addLoadingIndicator(loadingView, style: .fullView)
```

Or this for a centered example with custom size:
```swift
let loadingView = MediaCellProgressView(frame: .zero)
imageView.addLoadingIndicator(loadingView, style: .centered(withSize: CGSize(width: 60, height: 60)))
```
